src/panel/PanelManager.jsx

```diff
@@ -9,128 +8,81 @@ import DirectionPanel from 'src/panel/direction/DirectionPanel';
 import Telemetry from 'src/libs/telemetry';
 import CategoryService from 'src/adapters/category_service';
 import { parseQueryString, getCurrentUrl, buildQueryString } from 'src/libs/url_utils';
-import { fire, listen } from 'src/libs/customEvents';
+import { fire, listen, unListen } from 'src/libs/customEvents';
 import { isNullOrEmpty } from 'src/libs/object';
-import { isMobileDevice } from 'src/libs/device';
 import { PanelContext } from 'src/libs/panelContext.js';
 import NoResultPanel from 'src/panel/NoResultPanel';
 import TopBar from 'src/components/TopBar';
+import { useConfig, useDevice } from 'src/hooks';
 
-const directionConf = nconf.get().direction;
-
-export default class PanelManager extends React.Component {
-  static propTypes = {
-    router: PropTypes.object.isRequired,
-  };
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      ActivePanel: ServicePanel,
-      options: {},
-      panelSize: 'default',
-      isSuggestOpen: false,
-      appInputValue: '',
-      userInputValue: '',
-    };
-
-    this.mainSearchInputRef = React.createRef();
+function getTopBarAppValue({ poiFilters = {}, poi = {}, query } = {}) {
+  if (poi.name) {
+    return poi.name;
+  }
+  if (poiFilters.category) {
+    return CategoryService.getCategoryByName(poiFilters.category)?.getInputValue();
   }
```

Comment:
What if the category is not found ?  
For consistency, I think that an empty string should be used as a default here too, similarly to the next line.
(eg: http://localhost:3000/places/?type=invalidType could trigger a query to "undefined" after the input is focused)

Reply:
Sure. I've made the change, thanks.